src/MetadataShared/DataHelper.cs

```diff
@@ -314,19 +312,39 @@ private IEnumerable<EntityMetadata> GetRelationMetadata(IEnumerable<EntityMetada
             return logicalNames;
         }
 
+        internal Guid? GetActiveSolution()
+        {
+            var query = new QueryExpression("solution")
+            {
+                ColumnSet = new ColumnSet(),
+                Criteria = new FilterExpression()
+            };
+            query.Criteria.AddCondition(new ConditionExpression("uniquename", ConditionOperator.Equal, "active"));
+
+            
+
+            return service.RetrieveMultiple(query).Entities
+                .Select(e => e.Id).FirstOrDefault();
+        }
+
         internal IEnumerable<Entity> GetWorkflows() {
+            var activeSolutionId = GetActiveSolution();
```

Comment:
Maybe add a comment of why we are using active solution instead of the solution we target?